docs/hyperscript.md

```diff
@@ -278,6 +278,29 @@ function doSomething(e) {
 m("div", {onclick: doSomething})
 ```
 
+Mithril adds event listeners to the DOM using [EventTarget#addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener). This means you can provide a function (as above) or an [EventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventListener) object. For example:
+
+```javascript
+var clickListener = {
+	handleEvent: function(e) {
+		console.log(e)
+	}
+}
+
+m("div", {onclick: clickListener})
+```
```

Comment:
You could probably explain the usefulness of it better if you combined `handleEvent` with component state. For example:

```js
var Counter = {
    handleEvent: function (ev) {
        if (ev.target.textContent === "+") {
            this.count++
        } else {
            this.count--
        }
    },

    view: function () {
        return m("div", [
            m("button", {onclick: this}, "-"),
            m("span", this.count),
            m("button", {onclick: this}, "+"),
        ])
    }
}
```

This uses `textContent` to tell elements apart, but you could also use other things like `event.type` to tell events apart on single elements.

For non-closure components, it does a good job of getting complex logic out of the view without too much boilerplate or ceremony. The `handleEvent` callback acts almost like a reducer of sorts (to borrow a React/Redux term), just it uses mutation to update state instead of returning a new object. [I've used a similar trick in a pure DOM thing](https://github.com/isiahmeadows/color-picker/blob/master/color-picker.js#L314-L434), to reduce boilerplate and make the logic easier to follow.

Reply:
As discussed on Gitter, we decided to nix the extra explanation.

Reply:
I think the resolution is misleading — it suggests Mithril is offering some special feature here when in fact this is just the native behaviour of event handlers.